src/terms/nonlocal.jl

```diff
@@ -179,7 +179,7 @@ end
 Build form factors (Fourier transforms of projectors) for an atom centered at 0.
 """
 function build_form_factors(psp, G_plus_ks)
-    T = typeof(psp).parameters[1]
+    T = real(typeof(norm(first(G_plus_ks))))
```

Comment:
I don't think that's the right thing to do here. I bet this breaks forward-diff support in pseudos as they need to pick up the type from the pseudo type parameters.

Reply:
I think using the pseudo type was breaking some other forward-diff tests; I think the `Gplusk_vectors_cart` are dual-typed when the model is. Maybe we should look into passing a type to the pseudo parser?